torchtune/training/quantization.py

```diff
@@ -219,3 +223,22 @@ def swap_lora_linear_with_qat(
                 activation_qat_config,
                 weight_qat_config,
             )
+
+
+def convert_to_float8_training(model: torch.nn.Module) -> torch.nn.Module:
+    """
+    Prepare the model for float8 training by swapping all `torch.nn.Linear` with `Float8Linear`.
+    """
+    print("doing fp8 quantized training")
+    fp8_config = Float8LinearConfig(
+        enable_fsdp_float8_all_gather=True,
+        force_recompute_fp8_weight_in_bwd=True,
```

Comment:
if we are gating on PT nightlies we can delete this, as a change landed to PT core recently to make us not need this config option anymore.  We plan to delete this option in the future.